tests/tracing/processor/test_otel_metrics.py

```diff
@@ -0,0 +1,108 @@
+"""Tests for OtelSpanProcessor metrics export functionality."""
+
+import time
+
+import pytest
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import InMemoryMetricReader
+
+import mlflow
+
+
+@pytest.fixture
+def metric_reader():
+    """Create an in-memory metric reader for testing."""
+    reader = InMemoryMetricReader()
+    provider = MeterProvider(metric_readers=[reader])
+    metrics.set_meter_provider(provider)
+    yield reader
+    provider.shutdown()
+
+
+def test_metrics_export(monkeypatch, metric_reader):
+    """Test that metrics are exported with correct attributes."""
+    monkeypatch.setenv("OTEL_EXPORTER_OTLP_METRICS_ENDPOINT", "http://localhost:9090")
+    mlflow.set_experiment("test_experiment")
+
+    @mlflow.trace(span_type="CHAIN", name="parent")
+    def parent_function():
+        mlflow.update_current_trace({"env": "test", "version": "1.0"})
+        time.sleep(0.01)  # 10ms
+        return child_function()
+
+    @mlflow.trace(span_type="LLM", name="child")
+    def child_function():
+        time.sleep(0.25)  # 250ms
+        return "result"
+
+    @mlflow.trace(span_type="TOOL", name="error_function")
+    def error_function():
+        time.sleep(1.0)  # 1000ms
+        raise ValueError("Test error")
+
+    # Execute successful trace
+    parent_function()
+    # Execute error trace
+    with pytest.raises(ValueError, match="Test error"):
+        error_function()
+
+    metrics_data = metric_reader.get_metrics_data()
+    assert metrics_data is not None
+
+    data_points = []
+    for resource_metric in metrics_data.resource_metrics:
+        for scope_metric in resource_metric.scope_metrics:
+            for metric in scope_metric.metrics:
+                if metric.name == "mlflow.trace.span.duration":
+                    assert metric.unit == "ms"
+                    data_points.extend(metric.data.data_points)
+
+    assert len(data_points) == 3, "Expected exactly 3 span metrics"
```

Comment:
```suggestion
    assert len(data_points) == 3
```

this seems redundant.